resources/js/components/Search/Autocomplete.vue

```diff
@@ -12,43 +12,75 @@ export default {
             type: Number,
             default: 100,
         },
-    },
-
-    mounted() {
-        this.$nextTick(() => this.$emit('mounted'))
+        limit: {
+            type: Number,
+            default: 10,
+        },
+        multiMatchTypes: {
+            type: Array,
+            default: () => ['best_fields', 'phrase', 'phrase_prefix'],
+        },
     },
 
     render() {
-        return this.$scopedSlots.default(Object.assign(this, { self: this }))
+        return this.$scopedSlots.default(this)
     },
 
     data() {
         return {
-            results: { count: 0 },
+            results: {},
+            resultsCount: 0,
+            searchAdditionals: () => null,
         }
     },
 
-    methods: {
-        searchAdditionals(query) {
-            if (!this.additionals) {
+    mounted() {
+        this.$nextTick(() => this.$emit('mounted'))
+        let self = this
+
+        // Define function here to gain access to the debounce prop
+        this.searchAdditionals = useDebounceFn(function (query) {
```

Comment:
Are we sure we want to use Debounce? Or do we want to use throttle?
What does reactivesearch do?

Reply:
ReactiveSearch (at least the DataSearch component) also does a debounce, this mirrors the functionality so that the search queries are always executed at the same time.